backend/de.metas.monitoring/src/main/java/de/metas/monitoring/adapter/MicrometerPerformanceMonitoringService.java

```diff
@@ -132,4 +168,11 @@ private <V> V recordCallable(final Callable<V> callable, final ArrayList<Tag> ta
 			throw PerformanceMonitoringServiceUtil.asRTE(e);
 		}
 	}
+
+	private IAutoCloseable reduceDepth()
```

Comment:
pretty weird name....
usually when u have a method which returns a closeable the expectation is to do something in that method and return an autocloseable which reverses the effect.

Basically, to have the logic which does and the logic which undoes in the same place.

But in ur case, u refactored here only the undo logic... which makes the code harder to read and maintain.
In this case i suggest to undo logic in some finally block of caller.